code/modules/antagonists/revenant/revenant.dm

```diff
@@ -392,6 +393,9 @@
 	if(!Adjacent(A))
 		to_chat(src, "<span class='warning'>You can only orbit things that are next to you!</span>")
 		return
+	if(isobserver(A) || isrevenant(A) || istype(A, /mob/living/simple_animal/eminence))
```

Comment:
rather, check if the thing they are orbiting has an invisibility over INVISIBILITY_OBSERVER.

Reply:
The goal is stopping revs from orbiting something that can't be seen, right? Or is the problem that ghosts can change z levels?

Reply:
> Or is the problem that ghosts can change z levels?

correct. I believe other mobs using z-jaunt(even including hand tele or wizard teleport) is fine since theirs is limited, but ghost/eminence one would be bad as theirs is unlimited.
while revenant orbiting revenant would make a serious teaming up issue - as I know teamed revenants are really powerful.
So, that's why these are specifically blocked.

Reply:
maybe eminence might be not an issue since rev just can orbit cults then go to reebe togather.

Reply:
PR updated.
* Revenant can teleport to the station as long as they're not in the station
* eminence check is removed for that condition